src/before-all.ts

```diff
@@ -0,0 +1,50 @@
+import "reflect-metadata"
+import { TestOptionsInput, parseTestOptions, isTestOptionsInput } from "./options"
+import { configuration } from "./configure"
+
+export type BeforeAllFunction = (instance: any) => void
+
+export function getBeforeAlls(target: Object): BeforeAllFunction[] {
+    const beforeAlls = Reflect.getMetadata("test-decorator:beforeAll", target)
```

Comment:
in `test.ts` the name of the reflection key is plural, please name them consistently and either change `test-decorator:tests` to `test-decorator:test` or `test-decorator:beforeAll` to `test-decorator:beforeAlls`. Now that I am thinking about this, maybe we should rename the prefix `test-decorator` to `test-decorators` as it would then match the name of the package.